src/components/encoding-pane/encoding-shelf.tsx

```diff
@@ -28,6 +29,8 @@ export interface EncodingShelfPropsBase extends ActionHandler<ShelfEncodingActio
   id: ShelfId;
 
   fieldDef: ShelfFieldDef;
+
+  schema?: Schema;
```

Comment:
Oh but I should have set it to a required parameter, just don't call schema.domain to get the filters.